datahub-frontend/app/auth/sso/oidc/OidcAuthorizationGenerator.java

```diff
@@ -0,0 +1,45 @@
+package auth.sso.oidc;
+
+import java.util.Map.Entry;
+
+import org.pac4j.core.authorization.generator.AuthorizationGenerator;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.AttributeLocation;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.definition.ProfileDefinition;
+import org.pac4j.oidc.profile.OidcProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTParser;
+
+public class OidcAuthorizationGenerator implements AuthorizationGenerator {
```

Comment:
Can we name this class to reflect that its related to Parsing access tokens?

Reply:
We can also keep this name, but just pass in the "OidcConfigs" object in the constructor, and then decide whether to parse JWT access tokens inside of here!

Reply:
Passed in OidcConfigs in constructor